### ISSUE TITLE
Streaming WAV data

### ISSUE THREAD

**Issue by elementechemlyn** [joeweiss/birdnetlib#75]
Hi. Thanks for the work on this library - it's fantastic.

I have a raspberry pi zero out in the garden sending wav data back to a pi 4 running the library. Rather than copying files around, I made a few updates to the library so it can handle in memory buffers of WAV data. The changes let me run it in a simple TCP server that receives a constant stream of WAVs.

There is an example of how it works here:

https://github.com/elementechemlyn/birdnetlib/blob/wavbuffer/examples/simple_tcp_server.py

The changes are pretty rudimental but are non-breaking and have been working well for me over the last few months.  If you think they might be useful for others and would like a pull request then let me know.

Cheers,
E.

**Comment by joeweiss** [comment]
Hi @elementechemlyn. This is a delightful addition. I'd love a pull request if you have a chance. 

What do you think of pulling it out into a new `RecordingBuffer` class? The interface would look something like ...

```
recording = RecordingBuffer(
    analyzer,
    buffer,
    rate=rate,
    lat=35.4244,
    lon=-120.7463,
    date=datetime(year=2022, month=5, day=10),  # use date or week_48
    min_conf=0.25,
)
recording.analyze()
```

That would avoid the necessity for `Recording(analyzer, None)` and be more explicit to its capabilities.

Thoughts?

And thanks again, this is an outstanding addition to the library. I've spent the last 10 minutes furiously looking for my two dusty Pi Zeros ... :)

**Comment by elementechemlyn** [comment]
That seems sensible. I was trying to not duplicate anything but also not break anything.

An option might be to move the common Recording stuff into a base class and then have two subclasses - one for files (which would be called Recorder to avoid a breaking change but could be called RecordingFile to be more explicit) and one for buffers?
 
Off the top of my head I think the only difference would be the constructors and the read_audio_data function. 

A second option would be to just sub-class Recording but I think that might need some duplication in the constructor to work around the None path issue?

The second option seems less "intrusive" but the first seems neater and potentially sets things up for other "recording options" down the line?

**Comment by joeweiss** [comment]
> the first seems neater and potentially sets things up for other "recording options" down the line

Good point. A base class makes sense to me. What about `BaseRecording`?

BaseRecording

Recording(BaseRecording)
RecordingBuffer(BaseRecording)

If you're up for it, I would welcome the PR.

**Comment by joeweiss** [comment]
PR has been merged. Thanks @elementechemlyn! This will be published in release 0.9.0.
